pkg/apis/edge/v1alpha1/logical-cluster.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ClusterProviderType string
+
+const (
+	KindProviderType ClusterProviderType = "kind"
+)
+
+// LogicalCluster represents a cluster.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:resource:scope=Cluster,shortName=ecl
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type LogicalCluster struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// `spec` describes a cluster.
+	// +optional
+	Spec LogicalClusterSpec `json:"spec,omitempty"`
+
+	// `status` describes the status of the cluster object.
+	// +optional
+	Status LogicalClusterStatus `json:"status"`
+}
+
+// LogicalClusterSpec describes a cluster.
+type LogicalClusterSpec struct {
+	// ProviderType is the type of the provider of the cluster.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="cluster is immutable"
+	ProviderType ClusterProviderType `json:"ProviderType"`
+
+	// ProviderName is the name of the provider of the cluster.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="cluster is immutable"
+	ProviderName string `json:"ProviderName"`
```

Comment:
Surely additional information is generally needed beyond type and name, such as client credentials and network endpoint.  How do those come in?

Reply:
We are creating a separate resource for the provider.  The first drop is in https://github.com/kcp-dev/edge-mc/pull/566 and the full provider information is yet to be finalized.